dbt_semantic_interfaces/implementations/elements/dimension.py

```diff
@@ -37,6 +42,16 @@ class PydanticDimensionTypeParams(HashableBaseModel):
     validity_params: Optional[PydanticDimensionValidityParams] = None
 
 
+class PydanticDimensionConfig(HashableBaseModel, ProtocolHint[DimensionConfig]):
```

Comment:
If these protocols & implementations are all exactly the same structure, I think we should just use the `MetricConfig` protocol & implementation for all of them. Would be nice to remove the duplicate code if it's not needed. @WilliamDee would you foresee any problems with that?
It would be great if we could rename the class to just something more generic like `SemanticLayerElementConfig` or something if we're going to use it on all these elements. That would technically be a breaking change for DSI, but I just checked in dbt-core and it's not used anywhere. So it should be fine.
We would want to bump DSI to the next minor version if we do that. So if we do that, I can walk you through the steps for that.

Reply:
We could do that, but a foreseeable issue comes when or if the Config ever diverges for different elements. Then in that case it may be abit of a pain in the ass to change.

Reply:
actually, seems like core just uses `config` as some arbitrary dict, https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/contracts/graph/unparsed.py#L615 so i think it's fine to generalize it in DSI

Reply:
Okay, I made this update for the resources in this PR. I will make a follow-up PR to consolidate the other resources. I will wait on Grace's approval to merge this PR.